File: src/multishcript/guards.clj
(ns multishcript.guards
  (:require [babashka.fs :as fs]))

(defn guard-args
  [args]
  (when (not= 1 (count args))
    (println "Use only one arg, the path to the multishcript you want to run.")
    (System/exit 1)))

(defn guard-file-exists
  [file]
  (when (not (fs/exists? file))
    (println "File doesn't exist.")
    (System/exit 1)))

File: src/guards.ts
export const guardArgs = (args: string[]) => {
	if (args.length !== 1) {
		console.log(
			"Only use one arg, with the path to the multishcript you want to run.",
		)
		Deno.exit(1)
	}
}

export const guardFileExists = (file: string) => {
	try {
		Deno.statSync(file)
	} catch (err) {
		if (err instanceof Deno.errors.NotFound) {
			console.log("Multishcript file doesn't exist.")
			Deno.exit(2)
		}
	}
}
